dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -78,74 +62,241 @@ impl CommandDiscovery {
             }
         }
 
-        for path in paths {
-            if path.exists() && path.is_dir() {
-                for entry in path.read_dir().unwrap() {
-                    let entry = entry.unwrap();
-                    let path = entry.path();
-                    if path.is_file() {
-                        let file_name = path.file_name().unwrap().to_str().unwrap();
-                        let file_name_lowercase = file_name.to_lowercase();
-                        if file_name_lowercase.ends_with(".dsc.resource.json") ||
-                           file_name_lowercase.ends_with(".dsc.resource.yaml") ||
-                           file_name_lowercase.ends_with(".dsc.resource.yml") {
-                            let resource = match load_manifest(&path)
-                            {
-                                Ok(r) => r,
-                                Err(e) => {
-                                    if return_all_resources {
+        // remove duplicate entries to improve perf of resource search
+        let mut uniques = HashSet::new();
+        paths.retain(|e|uniques.insert((*e).clone()));
+
+        Ok(paths)
+    }
+}
+
+impl Default for CommandDiscovery {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ResourceDiscovery for CommandDiscovery {
+
+    #[allow(clippy::too_many_lines)]
+    fn list_available_resources(&mut self, type_name_filter: &str, adapter_name_filter: &str) -> Result<BTreeMap<String, DscResource>, DscError> {
+
+        debug!("Listing resources with type_name_filter/adapter_name_filter: {:?}/{:?}", type_name_filter, adapter_name_filter);
```

Comment:
I think you can just have the variables in the formatted string

```suggestion
        debug!("Listing resources with type_name_filter/adapter_name_filter: {type_name_filter}/{adapter_name_filter}");
```